client/components/SignUp.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+class SignUp extends Component {
+  constructor() {
+    super();
+
+    this.initialState = {
+      userName: '',
+      email: '',
+      phone: '',
+      password: ''
+    };
+
+    this.state = this.initialState;
+    this.handleChange = this.handleChange.bind(this);
+    this.registerUser = this.registerUser.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  registerUser(form) {
+    form.preventDefault();
+
+    const baseUrl = window.location.host;
+    const url = `http://${baseUrl}/api/v1/users/signup`;
+
+    axios.post(url, this.state)
```

Comment:
You should abstract this `axios` call to your actions alone, it should not be here in the `reducer`.